open_instruct/rejection_sampling/rejection_sampling.py

```diff
@@ -195,18 +245,30 @@ def main(args: Args):
     worst_offsets_per_model = {}
     reference_completion_scores_per_model = {}
     for model_name_or_path in args.model_names_or_paths:
-        with mp.Pool(args.num_gpus) as pool:
-            results = []
+        if "gpt-3.5" in model_name_or_path or "gpt-4" in model_name_or_path:
+            use_openai = True
+        else:
+            use_openai = False
+
+        results = []
+        if not use_openai:
+            with mp.Pool(args.num_gpus) as pool:
+                for i in range(args.num_gpus):
+                    results.append(pool.apply_async(process_shard, (i, model_name_or_path, args, shards[i])))
+        else:
             for i in range(args.num_gpus):
-                results.append(pool.apply_async(process_shard, (i, model_name_or_path, args, shards[i])))
+                results.append(process_shard_api(model_name_or_path, args, shards[i]))
```

Comment:
We should make a note that when using LLM as a judge, `num_gpus` doesn't actually use that many GPUs, instead it's just the number of shards.